### PR TITLE
Provide certificate information to policy rules

### PR DESCRIPTION
**Pull request by zregvart**
This expands the input file provided to Conftest by adding a `image` key, in addition to the existing `attestations` key. Which holds the `ref` and the `signatures key that in turn holds the `keyid`, `sig`, `certificate`, `chain` and `metadata` keys. For example, to access the certificate of the signer one could do something like: `input.image.signatures[_].certificate` in Rego.

The attestation signature information and metadata that was previously present in the output has been replaced with the image signature information.

Ref. https://issues.redhat.com/browse/HACBS-2075

### CODE REVIEW

File: acceptance/image/image.go [hunk 1]
```diff
@@ -68,13 +71,41 @@ const (
...
+
 type imageState struct {
-	Attestations map[string]string
-	Images       map[string]string
-	Signatures   map[string]string
+	AttestationSignatures map[string]Signature
+	Attestations          map[string]string
+	Images                map[string]string
+	ImageSignatures       map[string]Signature
+	Signatures            map[string]string
```
Comment (lcarva) [inline acceptance/image/image.go:87]: What's the difference between `Signatures` and `ImageSignatures` ?

File: acceptance/image/image.go [hunk 2]
```diff
@@ -229,9 +265,18 @@ func createAndPushAttestationWithPatches(ctx context.Context, imageName, keyName
...
 
-	ctx, err = storeAttestationSignatures(ctx, signedAttestation)
-	if err != nil {
+	if sig, err := unmarshallSignatures(signedAttestation); err != nil {
 		return ctx, err
+	} else {
+		state.AttestationSignatures[imageName] = Signature{
+			KeyID:     sig.KeyID,
+			Signature: sig.Sig,
+			// Metadata: map[string]string{ // add the metadata we don't have in the signature
```
Comment (lcarva) [inline acceptance/image/image.go:274]: Intentionally commented out?

File: acceptance/image/image.go [hunk 3]
```diff
@@ -343,6 +388,126 @@ func createAndPushKeylessImage(ctx context.Context, imageName string) (context.C
 		return ctx, err
 	}
 
+	var state *imageState
+	ctx, err = testenv.SetupState(ctx, &state)
+	if err != nil {
+		return ctx, err
+	}
+
+	if state.Images == nil {
```
Comment (lcarva) [inline acceptance/image/image.go:397]: `state.init()` ?

File: features/__snapshots__/task_validate_image.snap [hunk 4]
```diff
@@ -49,11 +49,7 @@ components:
   name: ""
   signatures:
   - keyid: ""
-    metadata:
```
Comment (lcarva) [inline features/__snapshots__/task_validate_image.snap:52]: We do still want this metadata, right?

**Review by lcarva — APPROVED** [review]

File: features/__snapshots__/task_validate_image.snap (hunk 4 above)
Comment (lcarva) [reply features/__snapshots__/task_validate_image.snap:52]: Yes, there's a follow up for this: https://issues.redhat.com/browse/HACBS-2296

File: internal/evaluation_target/application_snapshot_image/application_snapshot_image_test.go [hunk 5]
```diff
@@ -594,3 +620,56 @@ func TestValidateAttestationSignatureClaims(t *testing.T) {
...
...IyZmUz\nNDYzMTYwMDUzMWEwHQYKKwYBBAGDvzABCwQPDA1naXRodWItaG9zdGVkMDsGCisG\nAQQBg78wAQwELQwraHR0cHM6Ly9naXRodWIuY29tL2NoYWluZ3VhcmQtaW1hZ2Vz\nL2ltYWdlczA4BgorBgEEAYO/MAENBCoMKGUxZGNkZjcwYmUzMjZhNDk0Mjk1NzU0\nNjIyZmUzNDYzMTYwMDUzMWEwHwYKKwYBBAGDvzABDgQRDA9yZWZzL2hlYWRzL21h\naW4wGQYKKwYBBAGDvzABDwQLDAk1NjM1MTA5NTIwNAYKKwYBBAGDvzABEAQmDCRo\ndHRwczovL2dpdGh1Yi5jb20vY2hhaW5ndWFyZC1pbWFnZXMwGQYKKwYBBAGDvzAB\nEQQLDAkxMTMxOTg1NDUwagYKKwYBBAGDvzABEgRcDFpodHRwczovL2dpdGh1Yi5j\nb20vY2hhaW5ndWFyZC1pbWFnZXMvaW1hZ2VzLy5naXRodWIvd29ya2Zsb3dzL3Jl\nbGVhc2UueWFtbEByZWZzL2hlYWRzL21haW4wOAYKKwYBBAGDvzABEwQqDChlMWRj\nZGY3MGJlMzI2YTQ5NDI5NTc1NDYyMmZlMzQ2MzE2MDA1MzFhMBQGCisGAQQBg78w\nARQEBgwEcHVzaDBeBgorBgEEAYO/MAEVBFAMTmh0dHBzOi8vZ2l0aHViLmNvbS9j\naGFpbmd1YXJkLWltYWdlcy9pbWFnZXMvYWN0aW9ucy9ydW5zLzUxOTU1MDc2MzYv\nYXR0ZW1wdHMvMTCBigYKKwYBBAHWeQIEAgR8BHoAeAB2AN09MGrGxxEyYxkeHJln\nNwKiSl643jyt/4eKcoAvKe6OAAABiJPZADAAAAQDAEcwRQIgdHXB0QGS/GWkBnY1\nAZXSwb6/tbnnaVeWzde3t0fkkRMCIQC0bwdhWep548Cp4LzBPgGD0eioadqQdJHe\nXtVXBkD1dDAKBggqhkjOPQQDAwNoADBlAjBPpXDUSaAk5D6T1Eaqh+TRSQXr6rqV\nYxAJb/NgDbq8tTVLKustJDu2V9TQcpSzuKICMQDt0EAHmTISmKC8H3dciTrySh2l\nuS2rfl+L2AFS6DxAmVTBR3dlbrxQsUxshBWyH5s=\n-----END CERTIFICATE-----\n"
```
Comment (simonbaird) [inline internal/evaluation_target/application_snapshot_image/application_snapshot_image_test.go:634]: Consider using line breaks and backtick string delimiters here maybe.

**Review by simonbaird — APPROVED** [review]

File: acceptance/image/image.go (hunk 2 above)
Comment (zregvart) [reply acceptance/image/image.go:274]: Leftover from attestation signature/info

File: acceptance/image/image.go (hunk 1 above)
Comment (zregvart) [reply acceptance/image/image.go:87]: Added a comment

Comment (lcarva) [reply acceptance/image/image.go:93]: I still don't understand the difference between those two :sweat_smile:

**Review by lcarva — APPROVED** [review]

File: acceptance/image/image.go (hunk 1 above)
Comment (zregvart) [reply acceptance/image/image.go:93]: > I still don't understand the difference between those two sweat_smile

Fair, I'll reword/refactor this in a followup, this has been open for a minute now...

**Pull request merged**